website/docs/gen-ai/training/Llama2.md

```diff
@@ -0,0 +1,344 @@
+
+**MPI Operator:** An operator in Kubernetes is a method of packaging, deploying, and managing a Kubernetes application. The MPI Operator automates the deployment and management of MPI workloads.
+
+**FSx for Lustre:** A shared, high-performance filesystem which is well suited for workloads such as machine learning, high performance computing (HPC), video processing, and financial modeling. The FSx for Lustre filesystem will be shared across worker pods in the training job, providing a central repository to access the training data and to store model artifacts and logs.
+
+![Llama-2-trn1](img/llama2-trainium.png)
+
+## Deploying the Solution
+
+**Steps to train Llama-2 using AWS Trainium on Amazon EKS**
+
+Note: This post makes use of Meta’s Llama tokenizer, which is protected by a user license that must be accepted before the tokenizer files can be downloaded. Please ensure that you have access to the Llama files by requesting access here.
+
+<CollapsibleContent header={<h2><span>Prerequisites</span></h2>}>
+Before we begin, ensure you have all the prerequisites in place to make the deployment process smooth and hassle-free.
+Ensure that you have installed the following tools on your EC2 or Cloud9 instance.
+
+* [EC2 Instance](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EC2_GetStarted.html) or [Cloud9 instance](https://docs.aws.amazon.com/cloud9/latest/user-guide/tutorial-create-environment.html) → for both, please ensure you have 100GB+ of storage
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html)
+* [kubectl](https://Kubernetes.io/docs/tasks/tools/)
+* Git(Only for EC2 instance); Cloud9 comes with git installed by default
+* Docker
+* [terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+* Python, pip, jq, unzip
+
+To install all the pre-reqs on EC2, you can run this [script](https://github.com/sanjeevrg89/data-on-eks/blob/main/ai-ml/trainium-inferentia/examples/llama2/install-pre-requsites-for-ec2.sh) which is compatible with Amazon Linux 2023.
+
+
+Clone the Data on EKS repository
+
+```bash
+git clone https://github.com/awslabs/data-on-eks.git
+```
+
+Navigate to the trainium-inferentia directory.
+
+```bash
+cd data-on-eks/ai-ml/trainium-inferentia
+```
+
+Modify the **“trn1-32xl-ng1”** node group size in eks.tf file. Go to line 179 and change the min_size to 4, max_size to 4 and desired_size to 4.
+
+In addition, also update **variables.tf** for MPI operator to be installed. By default its not installed and for this post its important to change the default value from **false** to **true**
+
+Run the install script to provision an EKS cluster with all the add-ons needed for the solution.
+
+```bash
+./install.sh
+```
+
+
+### Verify the resources
+
+Verify the Amazon EKS Cluster
+
+```bash
+aws eks --region us-west-2 describe-cluster --name <cluster-name>
```

Comment:
change the cluster-name to the actual name defined in the varaibles

Reply:
The reason I cannot use the same name is I am appending a random string as discussed above